firewood/src/merkle/node.rs

```diff
@@ -440,10 +471,14 @@ impl Storable for Node {
             }
         };
 
+        let mut encoded_len: u64 = 0;
+        let mut encoded = None;
         if let Some(b) = self.encoded.get() {
             attrs.insert(if b.len() > TRIE_HASH_LEN {
                 NodeAttributes::LONG
             } else {
+                encoded_len = b.len() as u64;
+                encoded = Some(b);
```

Comment:
This code is now really hard to read because too many things are happening inside the same branching logic. Try to flatten things out. 

```rust
let (encoded_len, encoded) = if let Some(b) = self.encoded.get() {
    let (encoded_len, endoded, attribute) = if b.len() > TRIE_HASH_LEN {
        (0, None, NodeAttributes::LONG)
    } else {
        (b.len() as u64, Some(b), NodeAttributes::IS_ENCODED_BIG_VALID)
    };
    
    attrs.insert(attribute);
    
    (encoded_len, encoded)
} else if Some(b) = self.is_encoded_longer_than_hash_len.get() {
    attrs.insert(if *b {
        NodeAttributes::LONG
    } else {
        NodeAttributes::IS_ENCODED_BIG_VALID
    });
    
    (0, None)
} else {
    (0, None)
};
```

There are other ways to write this that might be better, but the point is that you should be setting a value while you're inserting another value. It might be clear right now, but it will be more difficult to change.

Reply:
@xinifinity ^^ I edited what I said